### ISSUE TITLE
Add "tips" for SVGP optimization in practice

### ISSUE THREAD

**Issue by FabricioArendTorres** [GPflow/GPflow#974]
Hi,

The documentation provides a basic overview of the SVGP, however in practice it does not seem to be that simple.

Depending on the problem, one comes across Github Issues like #820 that mention things like slowly increasing the gamma of the natural gradient and the corresponding paper.

Other unclear things are the choice of initial inducing points (as well as number of inducing points & minibatch size) - some papers mention the use of kmeans - or what you might recommend as stopping criteria. 

Would it be possible to provide an "In our experience..." section, that goes into those issues
and how you usually deal with them? 

That would be very helpful and awesome.

**Comment by hughsalimbeni** [comment]
On one hand I absolutely agree that more tips and tricks would likely be useful to users. However, there are a few potential dangers:

- The heuristics might actually be quite suboptimal, and elevating their status reduces the chance of finding this out.
- Heuristics might inappropriately become established as rules. 
- End users might stop thinking about important aspects of modelling and inference.

**Comment by FabricioArendTorres** [comment]
While I do get your point, I also think heuristics can serve as a good starting point.

It's not necessary that everyone starts again from 0, and this would make the 
package as well as scalable GPs more attractive to users - at least from a practical perspective.

Maybe an alternative would be a more extensive Example with a bigger (realistic) data set,
instead of toy experiments.
However, I guess that would be a lot more work.

**Comment by tadejkrivec** [comment]
Do you in general need much more iterations and optimization tuning to converge using SVGP compared to VFE? I generated some toy data for example and find it much harder to optimize for SVGP. The attached graphs of log likelihoods when optimizing with Adam are bellow. Likelihood is Gaussian and the whole training data is selected for batch.

### VFE
![vfe_gpflow](https://user-images.githubusercontent.com/44575819/59769320-30a72e80-92a6-11e9-980b-37f2aa2a972e.png)

### SVGP
![svgp_gpflow](https://user-images.githubusercontent.com/44575819/59769332-34d34c00-92a6-11e9-9bc9-71b589636ecd.png)

**Comment by markvdw** [comment]
The number of iterations needed by `SVGP` is sensitive to the learning rate. `SGPR` (which I assume you mean with `VFE`) does not have this problem as much, as it can be trained with BFGS variants.

Regarding documentation, I agree. We have limited time though, and it is not necessarily easy to distill "folk knowledge" into nice documentation. We do welcome external help.

If you're ever at conferences like NeurIPS/ICML/AISTATS, don't hesitate to drop by to chat. Papers and conference chats are a good way of exchanging folk wisdom :)

**Comment by st--** [comment]
@FabricioArendTorres Just to second that we would recommend help with documentation and tutorials! Feel free to add notebooks / documentation / etc in a PR to gpflow-develop (now based on tensorflow 2.0)!
